lib/webbluetooth/bindings.js

```diff
@@ -0,0 +1,298 @@
+  var self = this;
+  console.log('connect', deviceUuid);
+  var peripheral = this._peripherals[deviceUuid];
+
+  // Attempts to connect to remote GATT Server.
+  peripheral.device.connectGATT()
+    .then(function(gattServer){
+      peripheral.gattServer = gattServer;
+      console.log('peripheral connected', gattServer);
+      self.emit('connect', deviceUuid);
+    }, function(err){
+      console.log('err connecting', err);
+      self.emit('connect', deviceUuid, err);
+    });
+
+};
+
+NobleBindings.prototype.disconnect = function(deviceUuid) {
+  var peripheral = this._peripherals[deviceUuid];
+  if(peripheral.gattServer ){
+    peripheral.gattServer.disconnect();
+    this.emit('disconnect', deviceUuid);
+  }
+};
+
+NobleBindings.prototype.updateRssi = function(deviceUuid) {
+  var peripheral = this._peripherals[deviceUuid];
+
+  //TODO: need web api completed for this to work
+  this.emit('rssiUpdate', deviceUuid, rssi);
+};
+
+NobleBindings.prototype.discoverServices = function(deviceUuid, uuids) {
+  var peripheral = this._peripherals[deviceUuid];
+
+  //TODO: need web api completed for this to work
+  if(peripheral){
+    this.emit('servicesDiscover', deviceUuid, peripheral.device.uuids);
+  }
+
+};
+
+NobleBindings.prototype.discoverIncludedServices = function(deviceUuid, serviceUuid, serviceUuids) {
+  var peripheral = this._peripherals[deviceUuid];
+
+  //TODO impelment when web API has functionatility then emit response
+  //this.emit('includedServicesDiscover', deviceUuid, serviceUuid, includedServiceUuids);
+};
+
+NobleBindings.prototype.discoverCharacteristics = function(deviceUuid, serviceUuid, characteristicUuids) {
+  var peripheral = this._peripherals[deviceUuid];
+
+  //TODO need a web api to do this
+  if(peripheral){
+    this.emit('characteristicsDiscover', deviceUuid, serviceUuid, tempResponses.BEAN_CHARACTERISTICS[serviceUuid] || []);
+  }
+
+};
+
+NobleBindings.prototype._getPrimaryService = function(peripheral, serviceUuid){
+  serviceUuid = addDashes(serviceUuid);
+  
+  if(peripheral.cachedServices[serviceUuid]){
+    return new Promise(function(resolve, reject){
+      resolve(peripheral.cachedServices[serviceUuid]);
+    });
+  }
+  
+  return peripheral.gattServer.getPrimaryService(serviceUuid)
+    .then(function(service){
+      peripheral.cachedServices[serviceUuid] = service;
+      return service;
+    });
+};
+
+NobleBindings.prototype.read = function(deviceUuid, serviceUuid, characteristicUuid) {
+  var self = this;
+  var peripheral = this._peripherals[deviceUuid];
+  console.log('read', deviceUuid, serviceUuid, characteristicUuid);
+
+  self._getPrimaryService(peripheral, serviceUuid)
+    .then(function(service){
+      return service.getCharacteristic(addDashes(characteristicUuid));
+    })
+    .then(function(characteristic) {
+      return characteristic.readValue();
+    })
+    .then(function(data){
+      self.emit('write', peripheral.uuid, serviceUuid, characteristicUuid);
```

Comment:
This might be just me not understanding how noble works, but why do you emit a write here and below your error says 'error writing to a characteristic'? This is reading...